platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/turtle/runner/Turtle.java

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    /**
+     * Start the network. Does not simulate any time after each node is started.
+     */
+    public void start() {
+        for (final TurtleNode node : nodes) {
+            node.start();
+        }
+    }
+
+    /**
+     * Simulate the network for a period of time.
+     *
+     * @param duration the duration to simulate
+     */
+    public void simulateTime(@NonNull final Duration duration) {
+        final Instant simulatedStart = time.now();
+        final Instant simulatedEnd = simulatedStart.plus(duration);
+
+        while (time.now().isBefore(simulatedEnd)) {
+            reportThePassageOfTime();
+
+            time.tick(simulationGranularity);
+            network.tick(time.now());
+            tickAllNodes();
+        }
+    }
+
+    /**
+     * Generate a report of the passage of time.
+     */
+    private void reportThePassageOfTime() {
+        if (!timeReportingEnabled) {
+            return;
+        }
+        if (tickCount == 0) {
+            previousRealTime = Instant.now();
+            previousSimulatedTime = time.now();
+        }
+
+        if (tickCount > 0 && tickCount % 1000 == 0) {
+            final Instant now = Instant.now();
+
+            final Duration realElapsedTime = Duration.between(previousRealTime, now);
+            final Duration simulatedElapsedTime = Duration.between(previousSimulatedTime, time.now());
+
+            final double temporalVelocity;
+            if (realElapsedTime.isZero()) {
+                temporalVelocity = -1;
+            } else {
+                temporalVelocity = simulatedElapsedTime.toNanos() / (double) realElapsedTime.toNanos();
+            }
+
+            previousSimulatedTime = time.now();
+            previousRealTime = now;
+
+            System.out.printf(
+                    "tick %d, temporal velocity = %.2f, simulated time = %s%n",
+                    tickCount, temporalVelocity, time.now());
+        }
+
+        tickCount++;
+    }
+
+    /**
+     * Call tick() on all nodes in the network. Nodes do not interact with each other during their tick() phase, so it
+     * is safe to run them in parallel.
+     */
+    private void tickAllNodes() {
+        final List<Future<Void>> futures = new ArrayList<>();
+
+        // Iteration order over nodes does not need to be deterministic -- nodes are not permitted to communicate with
+        // each other during the tick phase, and they run on separate threads to boot.
+        for (final TurtleNode node : nodes) {
+            final Future<Void> future = threadPool.submit(() -> {
+                node.tick();
+                return null;
+            });
+            futures.add(future);
+        }
+
+        for (final Future<Void> future : futures) {
+            try {
+                future.get();
+            } catch (final InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
I would rethrow this exception, we want the test to fail loudly

Reply:
done